From 05860f041099fc1b9ecb2abc1236a9d80dccb80d Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 7 Jun 2017 17:55:34 +0800
Subject: [PATCH 1939/5242] MLK-15001-38 arm64: dtsi: fsl-imx8qxp: MIPI DSI PD
 should be contained by DC PD

commit  cc0c3ed5a5e13280b40fab96971a3deefca683b6 from
https://source.codeaurora.org/external/imx/linux-imx.git

The MIPI DSI power domains are the child power domains of DC power domain.
So, let's wrap the MIPI DSI power domains by DC power domain.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   99 ++++++++++--------------
 1 file changed, 43 insertions(+), 56 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 6ffd73b..7b50fd1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -590,72 +590,59 @@
 			};
 		};
 
-		pd_dc: dc-power-domain {
+		pd_dc0: PD_DC_0 {
 			compatible = "nxp,imx8-pd";
-			reg = <SC_R_LAST>;
+			reg = <SC_R_DC_0>;
 			#power-domain-cells = <0>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			pd_dc0: dc0-power-domain {
-				name = "dc0-power-domain";
-				reg = <SC_R_DC_0>;
+			pd_mipi_dsi0: PD_MIPI_0_DSI {
+				reg = <SC_R_MIPI_0>;
 				#power-domain-cells = <0>;
-				power-domains =<&pd_dc>;
-			};
-		};
-
-		pd_mipi_dsi0: PD_MIPI_0_DSI {
-			compatible = "nxp,imx8-pd";
-			reg = <SC_R_MIPI_0>;
-			#power-domain-cells = <0>;
-			#address-cells = <1>;
-			#size-cells = <0>;
+				power-domains =<&pd_dc0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
 
-			pd_mipi_dsi_0_i2c0: PD_MIPI_0_DSI_I2C0 {
-				name = "mipi0_dsi_i2c0";
-				reg = <SC_R_MIPI_0_I2C_0>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_mipi_dsi0>;
-			};
-			pd_mipi_dsi_0_i2c1: PD_MIPI_0_DSI_I2C1 {
-				name = "mipi0_dsi_i2c1";
-				reg = <SC_R_MIPI_0_I2C_1>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_mipi_dsi0>;
-			};
-			pd_mipi_0_pwm0: PD_MIPI_0_DSI_PWM0 {
-				name = "mipi0_dsi_pwm0";
-				reg = <SC_R_MIPI_0_PWM_0>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_mipi_dsi0>;
+				pd_mipi_dsi_0_i2c0: PD_MIPI_0_DSI_I2C0 {
+					reg = <SC_R_MIPI_0_I2C_0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_mipi_dsi0>;
+				};
+				pd_mipi_dsi_0_i2c1: PD_MIPI_0_DSI_I2C1 {
+					reg = <SC_R_MIPI_0_I2C_1>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_mipi_dsi0>;
+				};
+				pd_mipi_0_pwm0: PD_MIPI_0_DSI_PWM0 {
+					reg = <SC_R_MIPI_0_PWM_0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_mipi_dsi0>;
+				};
 			};
-		};
-
-		pd_mipi_dsi1: PD_MIPI_1_DSI {
-			compatible = "nxp,imx8-pd";
-			reg = <SC_R_MIPI_1>;
-			#power-domain-cells = <0>;
-			#address-cells = <1>;
-			#size-cells = <0>;
 
-			pd_mipi_dsi_1_i2c0: PD_MIPI_1_DSI_I2C0 {
-				name = "mipi1_dsi_i2c0";
-				reg = <SC_R_MIPI_1_I2C_0>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_mipi_dsi1>;
-			};
-			pd_mipi_dsi_1_i2c1: PD_MIPI_1_DSI_I2C1 {
-				name = "mipi1_dsi_i2c1";
-				reg = <SC_R_MIPI_1_I2C_1>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_mipi_dsi1>;
-			};
-			pd_mipi_1_pwm0: PD_MIPI_1_DSI_PWM0 {
-				name = "mipi1_dsi_pwm0";
-				reg = <SC_R_MIPI_1_PWM_0>;
+			pd_mipi_dsi1: PD_MIPI_1_DSI {
+				reg = <SC_R_MIPI_1>;
 				#power-domain-cells = <0>;
-				power-domains =<&pd_mipi_dsi1>;
+				power-domains =<&pd_dc0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_mipi_dsi_1_i2c0: PD_MIPI_1_DSI_I2C0 {
+					reg = <SC_R_MIPI_1_I2C_0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_mipi_dsi1>;
+				};
+				pd_mipi_dsi_1_i2c1: PD_MIPI_1_DSI_I2C1 {
+					reg = <SC_R_MIPI_1_I2C_1>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_mipi_dsi1>;
+				};
+				pd_mipi_1_pwm0: PD_MIPI_1_DSI_PWM0 {
+					reg = <SC_R_MIPI_1_PWM_0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_mipi_dsi1>;
+				};
 			};
 		};
 
-- 
1.7.9.5

